block/retriever.go

```diff
@@ -42,6 +42,9 @@ func (m *Manager) RetriveLoop(ctx context.Context) {
 // It fetches the batches from the settlement, gets the DA height and gets
 // the actual blocks from the DA.
 func (m *Manager) syncUntilTarget(ctx context.Context, syncTarget uint64) error {
+	m.executeBlockMutex.Lock()
```

Comment:
This process can be long as it needs to actually fetch the data from the DA upon new sync target (which can be dozens of seconds or even longer).
maybe I'm missing something but it looks like during this time no gossiped block won't be applied as the lock will block it (so basically rpc nodes won't provide "real-time" responses while syncing from the da) . 

I think the lock should be more fine grained on the actual execution of the blocks and not include the fetching from the da. 

Why are we not just putting it directly on the `applyBlock`/`executeBlock` function to only allow one caller access?

Reply:
aggreed. fetching data won't be mutex locked.

-----

> during this time no gossiped block won't be applied

gossiped block will be applied if it's height is correct, it quite conflicts with syncing process. I think it's fine for gossiped block to wait while syncing in progress.
in the happy flow there's no syncing anyway (as blocks are gossiped)

> directly on the applyBlock/executeBlock function 

The lock is more lock regarding the `store.Height()` and not only on execution

Reply:
Problem Is that blocks are synced also in the happy path with current design, or at least it should, in order to validate gossiped blocks against the da blocks, which is the source of truth. Though we didn't implement the validation yet (gossiped against da), I still think we should design a solution with this point in mind.

Reply:
still not sure I understand why can't `executeBlockMutex` can't live inside applyBlock?

the only operation I see that is being protected with current implementation is `m.store.NextHeight()`. 

If that's indeed the case I suggest changing the `nextHeight` or in general `height` access to be atomic and by that simplify the lock and only put it on the `applyBlock` function.